src/main/java/org/breedinginsight/daos/TraitDAO.java

```diff
@@ -554,9 +554,13 @@ private Trait parseTraitRecord(Record record, BiUserTable createdByUser, BiUserT
 
     private void saturateTrait(Trait trait, BrAPIObservationVariable brApiVariable) {
 
-        if (brApiVariable.getAdditionalInfo() != null && brApiVariable.getAdditionalInfo().has(TAGS_KEY)) {
-            List<String> tags = gson.fromJson(brApiVariable.getAdditionalInfo().getAsJsonArray(TAGS_KEY), List.class);
-            trait.setBrAPIProperties(brApiVariable, tags);
+        if (brApiVariable.getAdditionalInfo() != null &&
+                brApiVariable.getAdditionalInfo().has(TAGS_KEY)) {
+            // Prevents BrAPI server from failing the tags load
+            if (!brApiVariable.getAdditionalInfo().get(TAGS_KEY).isJsonNull()) {
```

Comment:
Does there need to be a check in the create/update calls so that null is not be written for the tags additional info value?

Reply:
Those work fine right now for saving a new trait with no tag and updating a trait to remove all of the tags, so I think we are good with leaving those alone.